datastore/sync_entity.go

```diff
@@ -263,6 +293,164 @@ func (dynamo *Dynamo) InsertSyncEntitiesWithServerTags(entities []*SyncEntity) e
 	return nil
 }
 
+// DisableSyncChain marks a chain as disabled so no further updates or commits can happen
+func (dynamo *Dynamo) DisableSyncChain(clientID string) error {
+	now := aws.Int64(utils.UnixMilli(time.Now()))
+	disabledMarker := SyncEntity{
+		ClientID:      clientID,
+		ID:            clientID,
```

Comment:
I wonder if we want to just use clientID as ID or define some tag to express that this is disabledMarker?
Using ID as clientID seems a bit wasted here since we can put more info into this, and we might have different tag item/type of marker in the future?

Reply:
Also, is every field in SyncEntity useful for disabledMarker? What's the reason of using SyncEntity for this rather than just defined another type that's more simple?

Reply:
The use of client id for both is due to how we have our table configured in dynamodb.  We have to provide the hash and range to query an individual record, which in our setup is client id and id.

We do a check in middleware on every request to verify the chain isn't deleted.  If it is we return the `DISABLED_BY_ADMIN` error code, which causes the clients to remove the chain.  This is how we propagate the delete out amongst clients.

Obviously if the client doesn't know the chain is deleted it couldn't know the id.  So we needed client id for both, which all clients know, so we can query for this marker.

As far as using a different type, I would just ask were you thinking of a separate table, or just a different type on the main table?  I know a lot of people use single table design with dynamo, which is recommended by AWS.  So, I assumed we were doing the same here.  I thought `SyncEntity` was generic enough for this use case.